graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/internal/ValidationErrorInstrumentation.kt

```diff
@@ -0,0 +1,34 @@
+package com.netflix.graphql.dgs.internal
+
+import com.netflix.graphql.dgs.exceptions.DgsException
+import com.netflix.graphql.types.errors.ErrorDetail
+import com.netflix.graphql.types.errors.ErrorType
+import com.netflix.graphql.types.errors.TypedGraphQLError
+import graphql.ExecutionResult
+import graphql.GraphQLError
+import graphql.execution.instrumentation.InstrumentationState
+import graphql.execution.instrumentation.SimplePerformantInstrumentation
+import graphql.execution.instrumentation.parameters.InstrumentationExecutionParameters
+import graphql.validation.ValidationError
+import java.util.concurrent.CompletableFuture
+
+class ValidationErrorInstrumentation : SimplePerformantInstrumentation() {
+    override fun instrumentExecutionResult(executionResult: ExecutionResult?, parameters: InstrumentationExecutionParameters?, state: InstrumentationState?): CompletableFuture<ExecutionResult> {
+        if (executionResult != null && executionResult.errors.isNotEmpty()) {
+            val newExecutionResult = ExecutionResult.newExecutionResult().from(executionResult)
+            val graphqlErrors: MutableList<GraphQLError> = mutableListOf()
+            executionResult.errors.filterIsInstance<ValidationError>().forEach { validationError ->
```

Comment:
are there other kind of errors we want to catch and transform into our `TypedGrahQLError` structure? For example, if this error [NonNullableFieldWasNullError](https://github.com/graphql-java/graphql-java/blob/master/src/main/java/graphql/execution/NonNullableFieldWasNullError.java#L44) is thrown will it be mapped correctly?

Reply:
it looks like on our v1 gateway side we used a [mapping](https://stash.corp.netflix.com/projects/ste/repos/graphlayer/browse/graphlayer-core/src/main/kotlin/com/netflix/graphlayer/core/TypedError.kt?at=c130399c3e4e599f79938752003787722f89e1f1#80) to transform errors thrown graphql-java style

Reply:
hmm..Ok, I might have to move this to the internal framework then. The framework doesn't really translate all  graphql-java errors to TypedGraphQLErrors as such.only exceptions are handled by default this way.